state_handlers.js

```diff
@@ -352,6 +361,37 @@ var scout_agent = (function() {
   };
 })();
 
+function matchArticleToTitlesHelper(stateObj) {
+  let searchTerm = getTitleFromSlotEvent(stateObj.event);
+  if (searchTerm) {
+    let thisVar = stateObj;
+    findBestScoringTitle(
+      searchTerm,
+      stateObj.attributes['titles'].articles
+    ).then(function(article) {
+      thisVar.attributes['chosenArticle'] = article.resolved_url;
+      thisVar.response
+        .speak(constants.strings.TITLE_CHOOSE_SUMM_FULL + article.title + '?')
+        .listen(constants.strings.TITLE_CHOICE_REPROMPT);
+      thisVar.emit(':responseReady');
+    });
+  }
+}
+
+function searchAndPlayArticleHelper(stateObj) {
+  if (stateObj.event.session.new) {
+    let stateObjThis = stateObj;
+    scout_agent.handleTitles(stateObj.event).then(function(titles) {
+      stateObjThis.attributes['titles'] = titles;
+      stateObjThis.handler.state = constants.states.TITLES_DECISION_MODE;
+      matchArticleToTitlesHelper(stateObjThis);
+    });
+  } else {
+    stateObj.handler.state = constants.states.TITLES_DECISION_MODE;
+    matchArticleToTitlesHelper(stateObj);
+  }
+}
+
```

Comment:
Suggestion from Slack thread:
![image](https://user-images.githubusercontent.com/2423092/39900554-6ccf497e-5477-11e8-9e61-281b0ee45863.png)

Reply:
Good suggestion. Changing.  I tested, it works.